pmd-core/etc/xslt/pmd-report-v2.xslt

```diff
@@ -0,0 +1,251 @@
+      <span class="navbar-toggler-icon"></span>
+    </button>
+  
+  </div>
+</nav>
+  <div class='container' style='margin-top:100px;'>
+    <h5>Problems found ::  <small> Version: <xsl:value-of select="$version"/> @ <xsl:value-of select="$timestamp"/></small>
+     <span id="summary"></span>  </h5>
+    <table id='pmdTable' class='table table-hover table-striped'>
+    <thead>
+ 
+      <tr>
+        <th>File #</th>
+        <th>Issue #</th>
+        <th>File</th>
+        <th>Line</th>
+        <th>Priority</th>
+        <th>Rule</th>
+        <th>Problem</th>
+
+   
+      </tr>
+      </thead>
+      <xsl:for-each select="scan:file">
+      <xsl:variable name="filename" select="@name" />
+      <xsl:variable name="slno" select="position()" />
+         <xsl:for-each select="scan:violation">
+          <xsl:variable name="pos" select="position()" />
+           <xsl:variable name="currPos" select="position()" />
+          <tr>
+          <td><xsl:value-of select="$slno"/></td>
+          <td><xsl:value-of select="position()"/></td>
+          <td><xsl:value-of select="$filename"/> </td>
+          <td><xsl:value-of select="@beginline"/></td>
+          <td><xsl:value-of select="@priority"/></td>
+          <xsl:variable name="priority" select="@priority" />
+         <script> total++; </script>
+    <xsl:choose>
+      <xsl:when test="$priority = 1">
+        <script> p1++; </script>
+      </xsl:when>
+         <xsl:when test="$priority = 2">
+        <script> p2++; </script>
+      </xsl:when>
+
+      <xsl:when test="$priority = 3">
+        <script> p3++; </script>
+      </xsl:when>
+      <xsl:when test="$priority = 4">
+        <script> p4++; </script>
+      </xsl:when>
+      <xsl:when test="$priority = 5">
+        <script> p5++; </script>
+      </xsl:when>
+
+
+      <xsl:otherwise>
+        <xsl:text></xsl:text>
+      </xsl:otherwise>
+    </xsl:choose>
+
+
+   
+
+
+    <xsl:variable name="ruleset" select="@ruleset" />
+    <xsl:choose>
+      <xsl:when test="$ruleset = 'Best Practices'">
+        <script> bp++; </script>
+      </xsl:when>
+         <xsl:when test="$ruleset = 'Code Style'">
+        <script> codeStyle++; </script>
+      </xsl:when>
+
+      <xsl:when test="$ruleset = 'Design'">
+        <script> design++; </script>
+      </xsl:when>
+
+        <xsl:when test="$ruleset = 'Documentation'">
+        <script> doc++; </script>
+      </xsl:when>
+
+      <xsl:when test="$ruleset = 'Error Prone'">
+        <script> ep++; </script>
+      </xsl:when>
+
+       <xsl:when test="$ruleset = 'Performance'">
+        <script> perf++; </script>
+      </xsl:when>
+
+       <xsl:when test="$ruleset = 'Security'">
+        <script> security++; </script>
+      </xsl:when>
+   
+
+      <xsl:otherwise>
+        <xsl:text></xsl:text>
+      </xsl:otherwise>
+    </xsl:choose>
+          <td><xsl:value-of select="@ruleset"/></td>
+
+          <td> <a href="{@externalInfoUrl}">Details</a></td>
```

Comment:
Nice, I didn't know this syntax. I assume, the XSL processor will replace this expression with the corresponding value.